manifests/server.pp

```diff
@@ -0,0 +1,77 @@
+# The class installs ClickHouse server and manages service
+#
+# @summary ClickHouse server class
+#
+# @param package_name
+#   Server package to be installed.
+# @param package_ensure
+#   Server package ensure. See `ensure` attribute for `package` resource.
+# @param service_name
+#   Name of the managed service for clickhouse-server.
+# @param service_ensure
+#   Desired state for `$service_name`, see `ensure` for `service` resource.
+# @param service_enable
+#   If `$service_name` should be enabled, see `enable` for `service` resource.
+# @param conf_dir
+#   Directory with clickhouse-server configuration.
+# @param conf_d_dir
+#   Directory with clickhouse-server included configuration.
+# @param users_d_dir
+#   Directory with clickhouse-server users configuration.
+# @param user
+#   User for configs owning. Strongly recommended staying default.
+#
+#   It's highly unrecommended to change default user, but feel free to shot in a knee.
+# @param group
+#   Group for configs, see `user` parameter.
+#
+# @example Simple use
+#   include clickhouse::server
+#
+# @example Use with params
+#   class { 'clickhouse::server':
+#     package_name   => 'clickhouse-server-custom',
+#     package_ensure => 'latest',
+#     service_name   => 'clickhouse-server',
+#     service_ensure => false,
+#     service_enable => false,
+#   }
+#
+# @author InnoGames GmbH
+#
+class clickhouse::server (
+    String[1]        $package_name   = 'clickhouse-server',
+    String[1]        $package_ensure = 'installed',
+    String[1]        $service_name   = $package_name,
+    Variant[Boolean, Enum[
+        'running',
+        'stopped'
+    ]]               $service_ensure = 'running',
+    Variant[Boolean, Enum[
+        'manual',
+        'mask'
+    ]]               $service_enable = true,
+    Stdlib::Unixpath $conf_dir       = '/etc/clickhouse-server',
+    Stdlib::Unixpath $conf_d_dir     = "${conf_dir}/conf.d",
+    Stdlib::Unixpath $users_d_dir    = "${conf_dir}/users.d",
+    String[1]        $user           = 'clickhouse',
```

Comment:
we might want to think about creating the system user if requested no?

Reply:
The default user is created from the package. If somebody would like to manage his own user, he would like to do it by itself.